tests/recovery/recovery_test/recovery_test.go

```diff
@@ -168,6 +172,172 @@ func TestSnapshotFixInvalidFiles(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TestConsistencyWhenKill9AfterModify will test the data consistency while it encountered kill -9 signal.
+func TestConsistencyWhenKill9AfterModify(t *testing.T) {
+	// assumption: the test is run on filesystem & not directly on object store
+	dataRepoPath := path.Join(*repoPathPrefix, dirPath, dataPath)
+
+	baseDir := t.TempDir()
+	if baseDir == "" {
+		t.FailNow()
+	}
+
+	bm, err := blobmanipulator.NewBlobManipulator(baseDir, dataRepoPath)
+	if err != nil {
+		if errors.Is(err, kopiarunner.ErrExeVariableNotSet) {
+			t.Skip("Skipping crash consistency tests because KOPIA_EXE is not set")
+		} else {
+			t.Skip("Error creating SnapshotTester:", err)
+		}
+	}
+
+	bm.DataRepoPath = dataRepoPath
+
+	// create a snapshot for initialized data
+	snapID, err := bm.SetUpSystemWithOneSnapshot()
+	if err != nil {
+		t.FailNow()
+	}
+
+	cmpDir := bm.PathToTakeSnapshot
+
+	copyDir := t.TempDir()
+	if copyDir == "" {
+		t.FailNow()
+	}
+
+	err = bm.FileHandler.CopyAllFiles(cmpDir, copyDir)
+	require.NoError(t, err)
+
+	// add files
+	fileSize := 1 * 1024 * 1024
+	numFiles := 200
+
+	err = bm.GenerateRandomFiles(fileSize, numFiles)
+	require.NoError(t, err)
+
+	err = bm.FileHandler.CopyAllFiles(bm.PathToTakeSnapshot, copyDir)
+	require.NoError(t, err)
+
+	// modify original files
+	content := "\nthis is a test for TestConsistencyWhenKill9AfterModify\n"
+	err = bm.FileHandler.ModifyDataSetWithContent(copyDir, content)
+	require.NoError(t, err)
+	log.Println("Copy content and modify the files.")
+
+	// kill the kopia command before it exits
```

Comment:
is this comment relevant here ?

You do not want to killl the `connect` command, but the snapshot command. Maybe move this comment to the snapshot command below ?

Reply:
Yes, you are right. I do the kill action before connect, so I should add connect with repository with the environment configuration, otherwise it will display "ERROR open repository: repository is not connected.kopia connect repo".